src/main/java/acecard/metrics/AcecardMetricsPlugin.java

```diff
@@ -166,49 +177,42 @@ public ClusterSettings getClusterSettings() {
 	// Listener Class for updating the documents
 	private class IndexListener implements IndexingOperationListener {
 		private String indexName;
-		private AcecardMetricsPlugin plugin;
 
-		private IndexListener(String indexName, AcecardMetricsPlugin plugin) {
+		private IndexListener(String indexName) {
 			this.indexName = indexName;
-			this.plugin = plugin;
 		}
 
 		@Override
 		public Engine.Index preIndex(ShardId shardId, Engine.Index index) {
-			if (plugin.isMasterNode.get()) {
-				String processorId;
-				String signalId;
 
-				List<LuceneDocument> docs = index.docs();
-				for (LuceneDocument d : docs) {
-					IndexableField processor = d.getField("processor");
-					IndexableField signal = d.getField("signal_id");
+			List<LuceneDocument> docs = index.docs();
+			for (LuceneDocument d : docs) {
+				String processorId = "UNDEFINED";
```

Comment:
Could we make this more generic in 2 ways.
1) Be able to define the field name in a config file and have it represented as "fieldX" in the code
2) Use this same genericization to allow for more than 2 (or rather "X" many) fields to be IndexableFields